### PR TITLE
API rewritten

### PR DESCRIPTION
**Pull request by Frenkiee**
- [ ] Backend: handle storing of single / multiple files in db
- [ ] Frontend: manual verification integration
- [ ] Frontend: handle multiple files in contract page

### CODE REVIEW

File: api/src/queries.ts [hunk 1]
```diff
@@ -0,0 +1,187 @@
...
+}
+
+const CONTRACT_VERIFICATION_STATUS = `SELECT status FROM contract_verification_request WHERE id = $1`;
+export const contractVerificationStatus = async (id: string): Promise<string> => {
+  const result = await query<Status>(CONTRACT_VERIFICATION_STATUS, [id]);
+  ensure(result.length > 0, "Contract does not exist...", 404);
+  return result[0].status;
+}
+
+const UPDATE_CONTRACT_STATUS = `UPDATE contract SET verified = TRUE, processed_bytecode = $1 WHERE contract_id = $2`;
```
Comment (mariopino) [inline api/src/queries.ts]: You need to also update in contract table:

- source: how we are going to handle multifile sources?
- compiler_version
- optimization
- runs
- target
- abi (from compilation!)
- license

NOTE: dont need to perform ERC20 token detection, that's handled by crawler.

File: api/src/index.ts [hunk 2]
```diff
@@ -0,0 +1,173 @@
...
+import { authenticationToken, config, getReefPrice, query } from './connector';
+import { checkIfContractIsVerified, contractVerificationInsert, contractVerificationStatus, findContractBytecode, findPool, findStakingRewards, findUserPool, findUserTokens, updateContractStatus } from './queries';
+import { AccountAddress, AppRequest, AutomaticContractVerificationReq, ContractVerificationID, License, ManualContractVerificationReq, PoolReq, UserPoolReq } from './types';
+import { ensure, ensureObjectKeys, errorStatus } from './utils';
+
+const cors = require('cors');
+const app = express();
+
+app.listen(config.httpPort, () => {
+  console.log(`Timezones by location application is running on port ${config.httpPort}.`);
```
Comment (mariopino) [inline api/src/index.ts]:
```suggestion
  console.log(`Reef explorer API is running on port ${config.httpPort}.`);
```

File: backend/docker/sql/backend.sql [hunk 3]
```diff
@@ -275,6 +275,7 @@ CREATE TABLE IF NOT EXISTS contract  (
   token_coinmarketcap_id TEXT DEFAULT NULL,
   -- token tracker
   timestamp BIGINT NOT NULL,
+  processed_bytecode TEXT DEFAULT NULL,
```
Comment (mariopino) [inline backend/docker/sql/backend.sql]: Consider adding a new colum `multifile_source` to handle contracts with multiple files.

**Review by mariopino — CHANGES_REQUESTED** [review]

File: backend/docker/sql/backend.sql (hunk 3 above)
Comment (Frenkiee) [reply backend/docker/sql/backend.sql]: no!

**Comment by mariopino** [comment]
@Frenkiee, seems some deps are missing:

`$ tsc && node ./dist/index.js
src/erc20Checker.ts(1,40): error TS2307: Cannot find module '@ethersproject/abi' or its corresponding type declarations.
src/queries.ts(6,35): error TS2307: Cannot find module 'ethers' or its corresponding type declarations.
error Command failed with exit code 2.
info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.
`

**Comment by mariopino** [comment]
Hi @Frenkiee: some contract columns are not being saved in db:

- source
- compiler_version
- optimization
- target

**Comment by Frenkiee** [comment]
Backend: handle storing of single / multiple files in db -> we are saving all the contracts in the same attribute, which is source! The logic for handling multiple sources is for the frontend to implement and you have it in chat from the last week.
Frontend: manual verification integration -> This is not for this branch to solve. Mario promised to implement it by the end of Friday last week.
Frontend: handle multiple files in contract page -> same as above.

`api/src/api.ts` we do not need it

**Comment by Netherdrake** [comment]
This PR is already huge, so merging into develop regardless of feature completeness to avoid work on stale branches.

Next PRs should be smaller, faster and more focused.

**Pull request merged**